### ISSUE TITLE
UnsupportedOperation seek while reading HTTP uri

### ISSUE THREAD

**Issue by laurentperrinet** [imageio/imageio#202]
I am using imageio for a while, but since I upgraded to python 3.6, I have problems loading an image from an online location

```
imageio.imread(url)
---------------------------------------------------------------------------
UnsupportedOperation                      Traceback (most recent call last)
<ipython-input-33-cd9354e362f7> in <module>()
----> 1 imageio.imread(url)

/usr/local/lib/python3.6/site-packages/imageio/core/functions.py in imread(uri, format, **kwargs)
    185     reader = read(uri, format, 'i', **kwargs)
    186     with reader:
--> 187         return reader.get_data(0)
    188
    189

/usr/local/lib/python3.6/site-packages/imageio/core/format.py in get_data(self, index, **kwargs)
    330             self._checkClosed()
    331             self._BaseReaderWriter_last_index = index
--> 332             im, meta = self._get_data(index, **kwargs)
    333             return Image(im, meta)  # Image tests im and meta
    334

/usr/local/lib/python3.6/site-packages/imageio/plugins/pillow.py in _get_data(self, index)
    300
    301         def _get_data(self, index):
--> 302             im, info = PillowFormat.Reader._get_data(self, index)
    303
    304             # Handle exif

/usr/local/lib/python3.6/site-packages/imageio/plugins/pillow.py in _get_data(self, index)
    113                     i += 1
    114                     self._seek(i)
--> 115             self._im.getdata()[0]
    116             im = pil_get_frame(self._im, self._grayscale)
    117             return im, self._im.info

/usr/local/lib/python3.6/site-packages/PIL/Image.py in getdata(self, band)
   1151         """
   1152
-> 1153         self.load()
   1154         if band is not None:
   1155             return self.im.getband(band)

/usr/local/lib/python3.6/site-packages/PIL/ImageFile.py in load(self)
    193                 decoder = Image._getdecoder(self.mode, decoder_name,
    194                                       args, self.decoderconfig)
--> 195                 seek(offset)
    196                 try:
    197                     decoder.setimage(self.im, extents)

UnsupportedOperation: seek

In [34]:
```

while a direct import would work:
```

import requests
from PIL import Image
url ='https://raw.githubusercontent.com/bicv/SLIP/master/database/serre07_targets/B_N107001.jpg'
im = Image.open(requests.get(url, stream=True).raw)
im.show()

```

my config is:

```
Python	3.6.0 64bit [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
IPython	5.1.0
OS	Darwin 16.3.0 x86_64 i386 64bit
numpy	1.12.0rc2
scipy	0.18.1
pillow	4.1.0.dev0
imageio	2.1.1dev
```

**Comment by almarklein** [comment]
Could it be that this has more to do with updating Pillow than updating Python? Perhaps Pillow added a `seek()` for some reason.

**Comment by laurentperrinet** [comment]
in the meanwhile, I am using this hack to handle HTTP links:

```
try:
    import imageio
    image = imageio.imread(URL)
except:
    from PIL import Image
    import requests
    from io import BytesIO
    response = requests.get(URL)
    image = np.array(Image.open(BytesIO(response.content)))
```

may be useful to others...

**Comment by almarklein** [comment]
@laurentperrinet note that you should use `except Exception:` instead of `except:`, because the latter also catches keyboard interrupts and system exit. In other words, if Python want to exit at the moment it is in the try clause, it would stop the exit procedure ...

**Comment by almarklein** [comment]
I can reproduce this. I tried reverting back the version of Pillow and of Python to see when the error was introduced but seem unable to do so. My guess is now that reading url's was actually broken from the start with the Pillow plugin, and we simply never realized it yet.

Anyway, the problem is that Pillow calls `tell()` on the file object. which fails. Fixing that, I found out that it *also* uses `seek()`, even though it seeks to the current position. I fixed it by "patching" the file object so that `tell()` works, and `seek()` wont fail if it seeks to the current pos.
